pkg/domain/infra/abi/cp.go

```diff
@@ -0,0 +1,435 @@
+// +build ABISupport
+
+package abi
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/buildah/pkg/chrootuser"
+	"github.com/containers/buildah/util"
+	"github.com/containers/libpod/libpod"
+	"github.com/containers/libpod/libpod/define"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/containers/storage"
+	"github.com/containers/storage/pkg/chrootarchive"
+	"github.com/containers/storage/pkg/idtools"
+	securejoin "github.com/cyphar/filepath-securejoin"
+	"github.com/docker/docker/pkg/archive"
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func (ic *ContainerEngine) ContainerCp(ctx context.Context, options entities.ContainerCpOptions) (*entities.ContainerCpReport, error) {
```

Comment:
```suggestion
func (ic *ContainerEngine) ContainerCp(ctx context.Context, source, destination string, options entities.ContainerCpOptions) (*entities.ContainerCpReport, error) {
```
Would be a signature more in-line with other methods